gm4_sweethearts/data/gm4_sweethearts/functions/transferring.mcfunction

```diff
@@ -1,12 +1,13 @@
 # @s = sneaking player holding flower
 # run from main
 
-tag @s add gm4_sweethearts_donor
+tag @s add gm4_sh_donor
 
-#get max health of player
-execute as @a[gamemode=!creative,gamemode=!spectator,distance=..3] store result score @s gm4_sh_data run attribute @s minecraft:generic.max_health get
+# get max and current health of players
+execute as @a[gamemode=!creative,gamemode=!spectator,distance=..3] run function gm4_sweethearts:get_health
 
 # transfer health to nearest player
-execute as @p[gamemode=!creative,gamemode=!spectator,tag=!gm4_sweethearts_donor,distance=..3,predicate=gm4_sweethearts:not_sneaking] unless score @s gm4_health >= @s gm4_sh_data at @s run function gm4_sweethearts:transfer_recipient 
+execute as @p[gamemode=!creative,gamemode=!spectator,tag=!gm4_sh_donor,distance=..3,predicate=gm4_sweethearts:not_sneaking] unless score @s gm4_sh_health >= @s gm4_sh_data at @s run function gm4_sweethearts:transfer_recipient 
 
-tag @s remove gm4_sweethearts_donor
+tag @s remove gm4_sh_donor
+scoreboard players reset @a gm4_sh_health
```

Comment:
Resetting this seems a bit pointless imo, especially since it adds an additional `@a`. `gm4_sh_data` doesn't get reset either and that's not a problem.